services/src/main/java/org/keycloak/broker/saml/mappers/AttributeToRoleMapper.java

```diff
@@ -115,21 +123,32 @@ public void importNewUser(KeycloakSession session, RealmModel realm, UserModel u
             user.grantRole(role);
         }
     }
-
+    
     protected boolean isAttributePresent(IdentityProviderMapperModel mapperModel, BrokeredIdentityContext context) {
-        String name = mapperModel.getConfig().get(ATTRIBUTE_NAME);
-        if (name != null && name.trim().equals("")) name = null;
-        String friendly = mapperModel.getConfig().get(ATTRIBUTE_FRIENDLY_NAME);
-        if (friendly != null && friendly.trim().equals("")) friendly = null;
+        String name = emptyToNull(mapperModel.getConfig().get(ATTRIBUTE_NAME));
+        String friendly = emptyToNull(mapperModel.getConfig().get(ATTRIBUTE_FRIENDLY_NAME));
         String desiredValue = mapperModel.getConfig().get(ATTRIBUTE_VALUE);
         AssertionType assertion = (AssertionType)context.getContextData().get(SAMLEndpoint.SAML_ASSERTION);
         for (AttributeStatementType statement : assertion.getAttributeStatements()) {
             for (AttributeStatementType.ASTChoiceType choice : statement.getAttributes()) {
                 AttributeType attr = choice.getAttribute();
-                if (name != null && !name.equals(attr.getName())) continue;
-                if (friendly != null && !name.equals(attr.getFriendlyName())) continue;
+                if (name != null) {
+                    String attributeName = emptyToNull(attr.getName()); // optional attribute
+                    
+                    if(attributeName == null || !name.equals(attributeName)) {
+                        continue;
+                    }
+                }
+                if (friendly != null) {
+                    String attributeFriendlyName = emptyToNull(attr.getFriendlyName()); // optional attribute
+                    if(attributeFriendlyName == null || !friendly.equals(attributeFriendlyName)) {
+                        continue;
+                    }
+                }
                 for (Object val : attr.getAttributeValue()) {
-                    if (val.equals(desiredValue)) return true;
+                    if (val != null && val.toString().equals(desiredValue)) {
```

Comment:
Test needs to be there specifically to show the need for "toString()" conversion.